basic/http/collect-user.js

```diff
@@ -25,13 +29,35 @@ class User {
 };
 
 const user = new User();
+
 router.get( '/users', async ctx => {
-  ctx.body = {data:user.data.map((e)=>{
+  ctx.body = {
+    data:user.data.map((e)=>{
+      return {id:e.id,name:e.name}; 
+    })
+  };
+});
+
+router.post('/users',async ctx => {
+  user.data.push(ctx.request.body);
+  
+  ctx.body = {
+    data:user.data.map((e)=>{
       return {id:e.id,name:e.name}; 
     })
   };
+});
 
+router.patch('/users',async ctx => {
+  user.data[1] = ctx.request.body;
```

Comment:
You're only getting the second element in the array of users.

You need to find the user given the placeholder of `:id`
Given this path, `router.patch('/users/:id' ...);`

Then the context should be like this:
```ts
ctx.params.id
```
Use `ctx.params.id` to find the user.

Reply:
RESOLVED 💯